src/app/(post-login)/submission/program/[shortName]/clinical-data/ClinicalEntityDataTable/index.tsx

```diff
@@ -0,0 +1,636 @@
+				? `${fieldName} is not a field within the latest dictionary. Please remove this from the ${entityName}.tsv file before submitting.`
+				: message;
+
+		const entries = errorGroup.length;
+
+		return {
+			entries,
+			fieldName,
+			entityName,
+			errorMessage,
+		};
+	});
+
+	const totalErrorCount = tableErrors.reduce(
+		(errorCount, errorGroup) => errorCount + errorGroup.entries,
+		0,
+	);
+
+	return [tableErrors, totalErrorCount];
+};
+
+const ClinicalEntityDataTable = ({
+	entityType,
+	program,
+	completionState = CompletionStates['all'],
+	currentDonors,
+	donorSearchResults = emptySearchResponse,
+	useDefaultQuery,
+	noData,
+}: {
+	entityType: string;
+	program: string;
+	completionState: CompletionStates;
+	currentDonors: number[];
+	donorSearchResults: ClinicalEntitySearchResultsQuery | ClinicalEntitySearchResultResponse;
+	useDefaultQuery: boolean;
+	noData: boolean;
+}) => {
+	const theme = useTheme();
+	const containerRef = createRef<HTMLDivElement>();
+
+	const defaultPageSettings =
+		useDefaultQuery && entityType === 'donor' ? defaultDonorSettings : defaultEntityPageSettings;
+	const [pageSettings, setPageSettings] = useState(defaultPageSettings);
+	const { page, pageSize, sorted } = pageSettings;
+
+	const [errorPageSettings, setErrorPageSettings] = useState(defaultErrorPageSettings);
+	const { page: errorPage, pageSize: errorPageSize, sorted: errorSorted } = errorPageSettings;
+	const { desc, id } = sorted[0];
+	const sortKey = aliasSortNames[id] || id;
+	const sort = `${desc ? '-' : ''}${sortKey}`;
+
+	const {
+		clinicalSearchResults: { searchResults, totalResults },
+	} = donorSearchResults || emptySearchResponse;
+
+	const nextSearchPage = (page + 1) * pageSize;
+
+	const donorIds = useDefaultQuery
+		? []
+		: currentDonors.length
+		? currentDonors
+		: searchResults.map(({ donorId }: ClinicalSearchResults) => donorId);
+
+	const submitterDonorIds =
+		useDefaultQuery || currentDonors.length
+			? []
+			: searchResults
+					.map(({ submitterDonorId }: ClinicalSearchResults) => submitterDonorId)
+					.filter((id) => !!id)
+					.slice(page * pageSize, nextSearchPage < totalResults ? nextSearchPage : totalResults);
+
+	const updatePageSettings = (key, value) => {
+		const newPageSettings = { ...pageSettings, [key]: value };
+
+		if (key === 'pageSize' && value !== pageSettings.pageSize) {
+			// Prevents bug querying nonexistent data
+			newPageSettings.page = 0;
+		}
+		setPageSettings(newPageSettings);
+		return newPageSettings;
+	};
+
+	useEffect(() => {
+		setPageSettings(defaultPageSettings);
+		setErrorPageSettings(defaultErrorPageSettings);
+	}, [entityType, useDefaultQuery]);
+
+	const { data: clinicalEntityData, loading } = useGetEntityData(
+		program,
+		entityType,
+		page,
+		pageSize,
+		sort,
+		completionState,
+		donorIds,
+		submitterDonorIds,
+	);
+
+	const { clinicalData } = clinicalEntityData;
+
+	// TODO: can you have ClinicalEntityErrors without ClinicalEntityData?
```

Comment:
Technically yes ClinicalEntityErrors is supposed to work as a separate standalone query, so this TODO can be removed.
However I tested and it is not working in Develop. I have a patch PR coming in shortly.